lib/patcher.js

```diff
@@ -123,23 +123,30 @@ function findReprints(newPath, reprints) {
     return canReprint;
 }
 
-function findAnyReprints(newPath, oldPath, reprints) {
+function findAnyReprints(newPath, oldPath, reprints, options) {
     var newNode = newPath.value;
     var oldNode = oldPath.value;
-
+        
+    if (options && options.quotemark &&
+          newPath.parentPath && 
+          newPath.parentPath.value &&
+          newPath.parentPath.value.type === "Literal" && 
+          isString.check(newPath.parentPath.value.value))
+        return false; //mark literal for reprint
```

Comment:
This is a bit aggressive: if the code wasn't modified by the traversal, then its quoting style should be left intact, I think. As long as you've implemented the printing code to use the preferred style for new nodes and nodes that need to be reprinted, I think you'll get the effect you want. You can always set `node.original = null` during the traversal to trigger reprinting for that node.